theseus_gui/src/store/search.js

```diff
@@ -0,0 +1,183 @@
+      ) {
+        activeCategories.forEach((cat) => formattedFacets.push(`categories:${cat}`))
+        activeCategories.forEach((cat) => formattedFacets.push(`categories:${cat}`))
+        activeLoaders.forEach((loader) => formattedFacets.push(`categories:${loader}`))
+        this.activeVersions.forEach((ver) => formattedFacets.push(`versions:${ver}`))
+
+        // If both are on or off, adding filters is pointless
+        if (this.environments.client === true && this.environments.server === false)
+          formattedFacets.push('client_side:required')
+        if (this.environments.client === false && this.environments.server === true)
+          formattedFacets.push('server_side:required')
+
+        if (this.openSource === true) formattedFacets.push('open_source:true')
+      }
+
+      let facets = '&facets=['
+      formattedFacets.forEach((facet) => (facets += `["${facet}"],`))
+      facets = facets.slice(0, facets.length - 1)
+      facets += ']'
+
+      let indexSort
+      switch (this.filter) {
+        case 'Download count':
+          indexSort = 'downloads'
+          break
+        case 'Follow count':
+          indexSort = 'follows'
+          break
+        case 'Recently published':
+          indexSort = 'newest'
+          break
+        case 'Recently updated':
+          indexSort = 'updated'
+          break
+        default:
+          indexSort = 'relevance'
+      }
+
+      return `?query=${this.searchInput || ''}&limit=${this.limit}&offset=${this.offset || 0}${
+        facets || ''
+      }&index=${indexSort}`
+    },
+    setSearchResults(response) {
+      this.searchResults = [...response.hits]
+      this.totalHits = response.total_hits
+      this.offset = response.offset
+      this.pageCount = Math.ceil(this.totalHits / this.limit)
+    },
+    setSearchInput(newInput) {
+      this.searchInput = newInput
+    },
+    setCurrentPage(newPage) {
+      this.currentPage = newPage
+
+      if (newPage === 1) this.offset = 0
+      else this.offset = this.currentPage * 10 - 10
+    },
+    setFilter(newFilter) {
+      this.filter = newFilter
+    },
+    toggleCategory(cat) {
+      this.categories[cat] = !this.categories[cat]
+    },
+    toggleLoader(loader) {
+      this.loaders[loader] = !this.loaders[loader]
+    },
+    toggleEnv(env) {
+      this.environments[env] = !this.environments[env]
+    },
+    setVersions(versions) {
+      this.activeVersions = versions
+    },
+    setLimit(newLimit) {
+      this.limit = newLimit
+    },
+    resetFilters() {
+      Object.keys(this.categories).forEach((cat) => {
+        this.categories[cat].enabled = false
+      })
+      Object.keys(this.loaders).forEach((loader) => {
+        this.loaders[loader].enabled = false
+      })
+      Object.keys(this.environments).forEach((env) => {
+        this.environments[env] = false
+      })
+      this.activeVersions = []
+    },
+  },
+  getters: {
+    getCategoriesByResultId: (state) => {
```

Comment:
this method doesn't need to exist, you can just pass in the generated.categories directly to the project card (it will handle the rest automatically